stdlib/FileWatching/src/pidfile.jl

```diff
@@ -192,23 +214,29 @@ function tryopen_exclusive(path::String, mode::Integer = 0o444)
     return nothing
 end
 
+struct PidfileLockFailure <: Exception
+    msg::String
+end
+
 """
-    open_exclusive(path::String; mode, poll_interval, stale_age) :: File
+    open_exclusive(path::String; mode, poll_interval, wait, wait_msg, stale_age) :: (File, waited::Bool)
```

Comment:
Hopefully changing what's returned is ok. This function isn't exported and I don't see any users https://juliahub.com/ui/Search?q=open_exclusive&type=code